packages/app/src/embed/components/Files/index.js

```diff
@@ -74,13 +70,86 @@ function Files({
           key={m.shortid}
           type={getType(m.title)}
           depth={depth}
-          setCurrentModule={setCurrentModule}
           active={m.id === currentModule}
           alternative={isMainModule(m, modules, directories, entry)}
+          onClick={() => setCurrentModule(m.id)}
         />
       ))}
     </Container>
   );
 }
 
+/** Utils to help identify module tree */
+
+const getCurrentModule = (modules, currentModuleId) =>
+  modules.find(module => module.id === currentModuleId);
+
+const getParentDirectory = (directories, child) =>
+  directories.find(directory => directory.shortid === child.directoryShortid);
+
+const getCurrentModuleTree = (directories, currentModule) => {
+  const currentModuleTree = [currentModule];
+
+  let parentDirectory = getParentDirectory(directories, currentModule);
+
+  while (parentDirectory) {
+    currentModuleTree.push(parentDirectory);
+    // get parent directory of the parent directory
+    parentDirectory = getParentDirectory(directories, parentDirectory);
+  }
+
+  return currentModuleTree;
+};
+
+function Directory({
+  directory,
+  currentModuleId,
+  setCurrentModule,
+  template,
+  entry,
+  modules,
+  directories,
+  depth,
+}) {
+  /** directory should be open by default if currentModule is inside it */
+
+  const currentModule = getCurrentModule(modules, currentModuleId);
+
+  const currentModuleTree = getCurrentModuleTree(directories, currentModule);
+
+  let openByDefault = false;
+  if (currentModuleTree.find(module => module.id === directory.id)) {
+    openByDefault = true;
+  }
+
+  const [open, setOpen] = React.useState(openByDefault);
```

Comment:
You can add the `openByDefault` calculation in the `useState`'s callback.

https://reactjs.org/docs/hooks-reference.html#lazy-initial-state

Reply:
Not sure if this counts as expensive, but can't hurt 😄

Reply:
Yeah I know it's not really expensive, but it makes it more readable and clear imo.